targets/simple_switch/primitives.cpp

```diff
@@ -473,6 +475,66 @@ class truncate_ : public ActionPrimitive<const Data &> {
 
 REGISTER_PRIMITIVE_W_NAME("truncate", truncate_);
 
+class log_msg : public ActionPrimitive<const std::string &, std::vector<Data>> {
+  void operator ()(const std::string &format, std::vector<Data> data_vector) {
+    std::string str = format;
+    std::string::size_type open = 0;  // position of the opened bracket
+    std::string::size_type end = 0;  // position of the matching closed bracket
+    unsigned int counter = 0;  // number occurrences of '{}' in string format
+    std::string::size_type tmp;
+
+    while ((open = str.find('{', open+1)) != std::string::npos) {
```

Comment:
I wonder if we can make that code less verbose by using C++11 regex, but not a strict requirement IMO